defaults/main.yml

```diff
@@ -41,16 +41,67 @@ docker__distribution_release: '{{ ansible_local.core.distribution_release
 docker__upstream: False
 
                                                                    # ]]]
+# .. envvar:: docker__variant [[[
+#
+# For upstream repositories the variant to be installed: ce or ee. Note that Docker EE
+# is not supported on Debian.
+docker__variant: 'ce'
+
+                                                                   # ]]]
+# .. envvar:: docker__channel [[[
+#
+# For upstream repositories choose the stable or edge channel.
+docker__channel: 'stable'
+
+                                                                   # ]]]
 # .. envvar:: docker__upstream_key [[[
 #
 # APT GPG key id used to sign the upstream Docker packages.
-docker__upstream_key: '58118E89F3A912897C070ADBF76221572C52609D'
+docker__upstream_key: '{{ "9DC858229FC7DD38854AE2D88D81803C0EBFCD88"
+                           if (docker__variant == "ce")
+                           else "DD911E995A64A202E85907D6BC14F10B6D085F96" }}'
+
+                                                                   # ]]]
+# .. envvar:: docker__version [[[
+#
+# The docker version to be installed. Leave empty to install the latest (stable or edge)
+# version, or specify a version. Example:
+#
+# .. code-block:: yaml
+#    :linenos:
+#
+#    docker__version: '17.05.0~ce-0~ubuntu-xenial'
+#
+docker__version: ''
+docker__packagename: '{{"docker-" + docker__variant + "=" + docker__version if docker__version|d() else "docker-" + docker__variant }}'
```

Comment:
This variable should have its own comment entry, and having it wrapped and indented would be better than using one long line, since the value is templated. Also, make sure to use spaces as separators between braces and data in the value.

Since this is used to specify upstream package name, I would rename this to variable to `docker__upstream_package_name`. Also see the previous comment for ideas about selecting a specific version; this shouldn't be done via Ansible `apt` module because it will result in unintended consequences later on.